client.go

```diff
@@ -112,14 +113,14 @@ func (r *client) processor() {
 				}
 				if !exists {
```

Comment:
Move the code from Line 107-121 into a helper function (or multiple, depending on how you want to break it up), since very similar logic is happening in Line 127-141. This will help with reducing redundancy and making the long nested conditionals/loops more readable.